src/main/java/org/fcrepo/audit/InternalAuditor.java

```diff
@@ -144,7 +144,7 @@ public void recordEvent(final FedoraEvent event) {
             isParentNodeLastModifiedEvent = true;
             final Iterator<String> propertiesIter = properties.iterator();
             String property;
-            while (properties.iterator().hasNext()) {
+            while (propertiesIter.hasNext()) {
```

Comment:
since `properties` is a `Set`, this would be better: `properties.stream().filter(...)`